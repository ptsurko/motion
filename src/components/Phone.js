import React, { PureComponent } from 'react';
import Types from 'prop-types';
import { buildTransform } from '../lib/utils';
import Coordinates from './Coordinates';
import PhoneSVG from './PhoneSVG';
import SatoriLogoSVG from './SatoriLogoSVG';
import AvatarSVG from './AvatarSVG';
import ArrowSVG from '../resources/Arrow.svg';

class Phone extends PureComponent {
  render() {
    const {
      isDefault,
      participant = { alpha: 0, beta: 0, gamma: 0 },
      showCoords = true,
      showRotation = true,
    } = this.props;
    const transform = (isDefault || !showRotation) ? '' : buildTransform(participant);

    return (
      <div className="phone-container">
        <div className="phone" style={{ transform }}>
          {isDefault
            ? <SatoriLogoSVG />
            : <AvatarSVG avatar={participant.avatar} color={participant.color} />
          }
          {(isDefault || !showRotation) && (
            <div>
              <img alt="rotation arrow 1" className="arrow arrow1" src={ArrowSVG} />
              <img alt="rotation arrow 2" className="arrow arrow2" src={ArrowSVG} />
            </div>
          )}

          <PhoneSVG />
        </div>
        {showCoords && <Coordinates participant={participant} />}
      </div>
    );
  }
};

Phone.propTypes = {
  isDefault: Types.bool,
  participant: Types.shape({
    alpha: Types.number.isRequired,
    beta: Types.number.isRequired,
    gamma: Types.number.isRequired,
  }),
  showCoords: Types.bool,
  showRotation: Types.bool,
};

export default Phone;
