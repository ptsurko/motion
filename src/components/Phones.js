import React, { PureComponent } from 'react';
import Types from 'prop-types';
import _ from 'lodash';
import Phone from './Phone';

class Phones extends PureComponent {
  render() {
    const { participants, showRotation } = this.props;

    return (
      <div className="phone-demo">
        {_.isEmpty(participants)
          ? <Phone isDefault />
          : _.map(participants, (participant, key) => <Phone key={key} participant={participant} showRotation={showRotation} />)
        }
      </div>
    );
  }
}

Phones.propTypes = {
  participants: Types.object.isRequired,
  showRotation: Types.bool,
};

export default Phones;
