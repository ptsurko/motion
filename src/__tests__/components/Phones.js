import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Phones from 'components/Phones';
import Phone from 'components/Phone';

it('renders Phones component correctly when there are no participants', () => {
  const Component = <Phones participants={{}} />;
  const tree = renderer.create(Component).toJSON();
  expect(tree).toMatchSnapshot();

  const wrapper = shallow(Component);
  expect(wrapper.find('Phone').equals(<Phone isDefault/>)).toBe(true);
});

it('renders Phones component correctly when there are participants', () => {
  const participants = {
    'abc': {
      alpha: -5,
      avatar: 'avatar5',
      beta: 2,
      color: '#88c440',
      deviceId: 'abc',
      gamma: 0,
      received_at: 1497911046581,
      roomId: '80ba136',
    },
  };
  const Component = <Phones participants={participants} />;
  const tree = renderer.create(Component).toJSON();
  expect(tree).toMatchSnapshot();

  const wrapper = shallow(Component);
  expect(wrapper.find('Phone').length).toBe(1);
});
