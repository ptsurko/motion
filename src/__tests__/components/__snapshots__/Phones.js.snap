// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Phones component correctly when there are no participants 1`] = `
<div
  className="phone-demo"
>
  <div
    className="phone-container"
  >
    <div
      className="phone"
      style={
        Object {
          "transform": "",
        }
      }
    >
      <svg
        className="phone-avatar"
        viewBox="0 0 36 31"
      >
        <title>
          Satori Logo
        </title>
        <defs>
          <polygon
            id="path-1"
            points="0 30.3545182 126.785352 30.3545182 126.785352 0.0182127109 0 0.0182127109"
          />
        </defs>
        <g
          fillRule="evenodd"
          id="Symbols"
          stroke="none"
          strokeWidth="1"
        >
          <g
            id="TopNavigation"
            transform="translate(-58.000000, -14.000000)"
          >
            <g
              id="Satori-Logo"
              transform="translate(58.000000, 14.000000)"
            >
              <polygon
                id="Fill-1"
                points="10.0869175 0.253030205 15.7789195 10.0309794 32.1506761 10.0309794 26.458674 0.253030205"
              />
              <polygon
                id="Fill-2"
                points="3.32435915 20.8113106 9.01636117 30.5892598 25.4134155 30.5892598 19.7214135 20.8113106"
              />
              <polygon
                id="Fill-3"
                points="8.79282072 0.253030205 4.40889316 7.76906187 -7.66599598e-05 15.3286017 17.5854626 15.3286017"
              />
              <polygon
                id="Fill-4"
                points="17.8896748 15.5127777 26.6823167 30.5883492 35.4749586 15.5127777"
              />
            </g>
          </g>
        </g>
      </svg>
      <div>
        <img
          alt="rotation arrow 1"
          className="arrow arrow1"
          src="Arrow.svg"
        />
        <img
          alt="rotation arrow 2"
          className="arrow arrow2"
          src="Arrow.svg"
        />
      </div>
      <svg
        className="phone-img"
        id="Layer_1"
        style={
          Object {
            "enableBackground": "new 0 0 162.6 332.5",
          }
        }
        version="1.1"
        viewBox="0 0 162.6 332.5"
        x="0px"
        y="0px"
      >
        <path
          d="M152.5,39.9v254.5H9.7V39.9H152.5 M154,38.5H8.3v257.4H154V38.5z M143.5,332.5H19.3 C8.8,332.6,0.1,324.1,0,313.6V19C0.1,8.4,8.7-0.1,19.3,0h124.2c10.5,0,19,8.5,19,19v294.6C162.5,324,154,332.5,143.5,332.5 L143.5,332.5z M19.3,2.5C10.1,2.4,2.6,9.8,2.5,19v294.6c0.1,9.2,7.6,16.5,16.8,16.4h124.2c9.1,0,16.5-7.3,16.6-16.4V19 c0-9.1-7.4-16.5-16.6-16.5H19.3z"
        />
      </svg>
    </div>
    <section
      className="phone-coords"
    >
      <div>
        <span>
          X: 
        </span>
        <span>
          0
        </span>
      </div>
      <div>
        <span>
          Y: 
        </span>
        <span>
          0
        </span>
      </div>
      <div>
        <span>
          Z: 
        </span>
        <span>
          0
        </span>
      </div>
    </section>
  </div>
</div>
`;

exports[`renders Phones component correctly when there are participants 1`] = `
<div
  className="phone-demo"
>
  <div
    className="phone-container"
  >
    <div
      className="phone"
      style={
        Object {
          "transform": "rotateZ(5deg) rotateX(2deg) rotateY(0deg)",
        }
      }
    >
      <div
        className="phone-avatar"
        onClick={[Function]}
      >
        <svg
          dangerouslySetInnerHTML={
            Object {
              "__html": "<title>5</title><g fill=\\"none\\" fill-rule=\\"evenodd\\"><path d=\\"M50 99.01c-27.614 0-50-22.163-50-49.505C0 22.164 22.386 0 50 0s50 22.164 50 49.505c0 27.342-22.386 49.505-50 49.505z\\" id=\\"bg\\" fill=\\"#88c440\\"></path><path d=\\"M58.836 35.873a3.828 3.828 0 0 1-3.826-3.83 3.828 3.828 0 0 1 3.826-3.83 3.816 3.816 0 0 1 3.57 2.483 2.489 2.489 0 0 0-2.183-1.305c-1.38 0-2.5 1.12-2.5 2.504a2.501 2.501 0 0 0 2.5 2.502 2.494 2.494 0 0 0 2.38-1.776c-.28 1.839-1.851 3.252-3.767 3.252m12.808-7.66c1.47 0 2.73.837 3.37 2.054a2.483 2.483 0 0 0-1.884-.876c-1.38 0-2.5 1.12-2.5 2.504a2.501 2.501 0 0 0 2.5 2.502c.936 0 1.742-.522 2.17-1.284-.464 1.59-1.916 2.76-3.656 2.76a3.827 3.827 0 0 1-3.824-3.83 3.828 3.828 0 0 1 3.824-3.83m-4.826 14.424c5.617 6.389 14.918 3.992 20.182 1.87-.87-8.947-9.036-17.124-9.036-17.124-18.653-5.375-30.786-5.09-38.676-2.683-.033-.155-.056-.25-.056-.25s8.017-7.78-1.907-7.438c0 0-5.245.473-.798 7.583.204.322.39.607.567.874-2.537 1-4.541 2.233-6.145 3.53.09-3.644-.581-10.876-8.252-4.232 0 0-4.715 4.566 4.578 8.145-1.999 2.875-2.412 5.265-2.412 5.265.114.885.195 1.738.294 2.608-2.519-4.1-7.564-9.863-11.306 1.644 0 0-2.277 8.598 11.543 5.64.154-.032.286-.065.434-.097.348 4.963.444 9.628.327 14.023-2.151-2.333-14.572-14.996-14.926 3.41 0 0 .385 10.356 14.602 2.786-.19 2.442 1.584 4.738 2.546 6.806 4.422 9.514 9.726 4.194 12.988-.087 4.91-6.447 12.835-8.863 12.835-8.863 24.25-2.914 31.887-10.62 32.783-18.391-20.165 4.694-20.165-5.02-20.165-5.02\\" fill=\\"#FFF\\"></path></g>",
            }
          }
          viewBox="0 0 100 100"
          width={undefined}
        />
      </div>
      <svg
        className="phone-img"
        id="Layer_1"
        style={
          Object {
            "enableBackground": "new 0 0 162.6 332.5",
          }
        }
        version="1.1"
        viewBox="0 0 162.6 332.5"
        x="0px"
        y="0px"
      >
        <path
          d="M152.5,39.9v254.5H9.7V39.9H152.5 M154,38.5H8.3v257.4H154V38.5z M143.5,332.5H19.3 C8.8,332.6,0.1,324.1,0,313.6V19C0.1,8.4,8.7-0.1,19.3,0h124.2c10.5,0,19,8.5,19,19v294.6C162.5,324,154,332.5,143.5,332.5 L143.5,332.5z M19.3,2.5C10.1,2.4,2.6,9.8,2.5,19v294.6c0.1,9.2,7.6,16.5,16.8,16.4h124.2c9.1,0,16.5-7.3,16.6-16.4V19 c0-9.1-7.4-16.5-16.6-16.5H19.3z"
        />
      </svg>
    </div>
    <section
      className="phone-coords"
    >
      <div>
        <span>
          X: 
        </span>
        <span>
          2
        </span>
      </div>
      <div>
        <span>
          Y: 
        </span>
        <span>
          0
        </span>
      </div>
      <div>
        <span>
          Z: 
        </span>
        <span>
          -5
        </span>
      </div>
    </section>
  </div>
</div>
`;
